### DISCUSSION TITLE
Instantiating Vanilla JS Class in SolidJs

### DISCUSSION THREAD

**Discussion by 99bits** [solidjs/solid#901] [General]
Hi Fellow SolidJs friends,

I am new to SolidJs and would like to use vanilla JS components in a SolidJs project.  
For example this calendar:  

https://mcdatepicker.netlify.app/usage

<div class="form-field">
  <label class="form-field__label">Datepicker Demo</label>
  <input id="datepicker" type="text">
</div>

/* javascript */
const datePicker = MCDatepicker.create({
  el: '#datepicker',
  bodyType: 'modal'
});

**Comment by orenelbaum** [comment]
Integrating things with Solid is always the same in Solid whether you want to integrate with classes or anything else. You can't use them directly. Unless you have a more specific question it's hard to describe the whole process of integrating a vanilla JS library with Solid, but in order to get a general idea maybe you can look at how other people have already integrated vanilla libraries.

**Comment by ryansolid** [comment]
Typically you wrap these things in component wrappers. If you can pass the element in, instead of a selector is better. But generally you'd do something like:

```js
function DatePicker() {
  onMount(() => {
    const datePicker = MCDatepicker.create({
      el: '#datepicker',
      bodyType: 'modal'
    });
    onCleanup(() => datePicker.dispose());
  });
  return <div id="datepicker" />
}
```
This is just a rough example. As I said getting a ref to the element would be better.

**Reply by trusktr** [reply]
Then if you want to map props to it, you can use an effect instead of onMount:

```js
function DatePicker(props) {
  createEffect(() => {
    const datePicker = MCDatepicker.create({
      el: '#datepicker',
      bodyType: props.bodyType // any time props.bodyType changes, the old picker is disposed, and a new one created
    });
    onCleanup(() => datePicker.dispose());
  });
  return <div id="datepicker" />
}
```

Note, if there's a way to update properties of `MCDatepicker`, then do that, instead of making a new one each time, as it would be more efficient, but the above at least would work.
